File: prefixSum/2003.py
n, m = map(int, input().split())
nums = list(map(int, input().split()))

start, end = 0, 0
total = 0
count = 0

while True:
    if total >= m:
        total -= nums[start]
        start += 1
    elif end == n:
        break
    else:
        total += nums[end]
        end += 1

    if total == m:
        count += 1

print(count)

File: prefixSum/2003.js
const fs = require("fs");

const [nmLine, numsLine] = fs
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n");
const [n, m] = nmLine.split(" ").map(Number);
const nums = numsLine.split(" ").map(Number);

let start = 0,
  end = 0;
let total = 0;
let count = 0;

while (true) {
  if (total >= m) {
    total -= nums[start];
    start++;
  } else if (end === n) {
    break;
  } else {
    total += nums[end];
    end++;
  }

  if (total === m) {
    count++;
  }
}

console.log(count);
